src/dash-table/utils/TableClipboardHelper.ts

```diff
@@ -5,15 +5,20 @@ import Clipboard from 'core/Clipboard';
 import Logger from 'core/Logger';
 
 import { ICellCoordinates, Columns, Data, SelectedCells } from 'dash-table/components/Table/props';
+import { createHeadings } from 'dash-table/components/Export/utils';
 import applyClipboardToData from './applyClipboardToData';
+import getHeaderRows from 'dash-table/derived/header/headerRows';
 
 export default class TableClipboardHelper {
     private static lastLocalCopy: any[][] = [[]];
+    private static localCopyWithoutHeaders: any[][] = [[]];
 
     public static toClipboard(e: any, selectedCells: SelectedCells, columns: Columns, data: Data) {
         const selectedRows = R.uniq(R.pluck('row', selectedCells).sort((a, b) => a - b));
         const selectedCols: any = R.uniq(R.pluck('column', selectedCells).sort((a, b) => a - b));
 
+        const transposedHeaders = createHeadings(R.pluck('name', columns), getHeaderRows(columns));
```

Comment:
When merging in with the hidden columns PR #508, will need to pass both the columns and visibleColumns -- visibleColumns for processing the data / creating the copy/paste info and columns (containing the hidden ones) to determine the number of header rows correctly